scripts/bump_major_version.py

```diff
@@ -0,0 +1,335 @@
+    # Int Int Path -> IO!
+
+    old_image_string = '''img src="{{ 'images/''' + \
+        version_string_name(old_version)
+    new_image_string = '''img src="{{ 'images/''' + \
+        version_string_name(new_version)
+
+    # Read in the file
+    filedata = None
+    try:
+        with open(file, 'r') as f:
+            filedata = f.read()
+    except UnicodeDecodeError:
+        return
+
+    # Replace the target string
+    filedata = filedata.replace(old_image_string, new_image_string)
+
+    # Write the file out again
+    with open(file, 'w') as f:
+        f.write(filedata)
+
+
+def grep_for_old_versionish_things(old_version, file):
+    # Int Path -> IO!
+    pat = re.compile(version_string_name(old_version))
+    lineno = 0
+    try:
+        with open(file) as f:
+            for line in f:
+                lineno += 1
+                matches = re.findall(pat, line)
+                if len(matches) != 0:
+                    print("{}:{}".format(file, lineno))
+                    print("{}".format(line))
+    except UnicodeDecodeError:
+        pass
+
+
+def update_base_yaml_config(to_version):
+    # Int -> IO!
+    file = os.path.join(docs_directory, '_config_base.yml')
+
+    # Read in the file
+    filedata = None
+    try:
+        with open(file, 'r') as f:
+            filedata = f.read()
+    except UnicodeDecodeError:
+        return
+
+    config = yaml.load(filedata, Loader=yaml.BaseLoader)
+
+    new_version_name = version_string_name(to_version)
+
+    newdata = '''
+    name: {}
+    version: {}
+    docker_image: cockroachdb/cockroach-unstable
+    build_time: 2020/05/12 11:00:26 (go1.13.4)
+    start_time: 2020-05-12 11:01:26.34274101 +0000 UTC
+'''.format(new_version_name, new_version_name)
+
+    newdata_loaded = yaml.load(newdata, Loader=yaml.BaseLoader)
+
+    config['release_info'][new_version_name] = newdata_loaded
+
+    # Write the file out again
+    with open(file, 'w') as f:
+        f.write(yaml.dump(config))
+
+
+def update_cockroachdb_yaml_config(from_version, to_version):
+    # Int Int -> IO!
+    file = os.path.join(docs_directory, '_config_cockroachdb.yml')
+
+    # Read in the file
+    filedata = None
+    try:
+        with open(file, 'r') as f:
+            filedata = f.read()
+    except UnicodeDecodeError:
+        return
+
+    config = yaml.load(filedata, Loader=yaml.BaseLoader)
+
+    old_version_name = version_string_name(from_version)
+    new_version_name = version_string_name(to_version)
+
+    config['versions']['stable'] = old_version_name
+    config['versions']['dev'] = new_version_name
+
+    # Write the file out again
+    with open(file, 'w') as f:
+        f.write(yaml.dump(config))
+
+
+def copy_releases_page(from_version, to_version):
+    # Int Int -> IO! State!
+    old_releases_page = os.path.join(docs_directory, 'releases',
+                                     version_string_name(from_version) +
+                                     '.0.md')
```

Comment:
Maybe a dumb question: Do we never do something like `from_version` "v19.2.6" --> `to_version` "v19.2.7"?

Reply:
Good question! We should never need to do that.  This script is only appropriate for use in the case when we do a major version bump, e.g. 20.1 -> 20.2.  This is because a new major release is the only time when we need to create a new major version folder and copy a bunch of stuff into it.

The case you mention where 'v19.2.6' -> 'v19.2.7' involves a human tech writer editing the YAML file during a point release.

But I take this convo to mean that this script needs additional docs, so I will work on that.

Reply:
Got it, thanks!

Reply:
Added some more (and hopefully, better) notes in the introductory doc/comment that I think answer these questions.  PTAL.